src/core/dyad_core.c

```diff
@@ -565,6 +649,96 @@ dyad_rc_t dyad_init (bool debug,
     return DYAD_RC_OK;
 }
 
+dyad_rc_t dyad_init_env (dyad_ctx_t **ctx)
+{
+    char *e = NULL;
+    bool debug = false;
+    bool check = false;
+    bool shared_storage = false;
+    unsigned int key_depth = 0;
+    unsigned int key_bins = 0;
+    char *kvs_namespace = NULL;
+    char *prod_managed_path = NULL;
+    char *cons_managed_path = NULL;
+    size_t dtl_mode_env_len = 0;
+    dyad_dtl_mode_t dtl_mode = DYAD_DTL_UCX;
+
+    if ((e = getenv (DYAD_SYNC_DEBUG_ENV))) {
+        debug = true;
+        enable_debug_dyad_utils ();
+    } else {
+        debug = false;
+        disable_debug_dyad_utils ();
+    }
+
+    if (debug)
+        fprintf (stderr, "DYAD_CORE: Initializing with environment variables\n");
+
+    if ((e = getenv (DYAD_SYNC_CHECK_ENV))) {
+        check = true;
+    } else {
+        check = false;
+    }
+
+    if ((e = getenv (DYAD_SHARED_STORAGE_ENV))) {
+        shared_storage = true;
+    } else {
+        shared_storage = false;
+    }
+
+    if ((e = getenv (DYAD_KEY_DEPTH_ENV))) {
+        key_depth = atoi (e);
+    } else {
+        key_depth = 3;
+    }
+
+    if ((e = getenv (DYAD_KEY_BINS_ENV))) {
+        key_bins = atoi (e);
+    } else {
+        key_bins = 1024;
+    }
+
+    if ((e = getenv (DYAD_KVS_NAMESPACE_ENV))) {
+        kvs_namespace = e;
+    } else {
+        kvs_namespace = NULL;
+    }
+
+    if ((e = getenv (DYAD_PATH_CONSUMER_ENV))) {
+        cons_managed_path = e;
+    } else {
+        cons_managed_path = NULL;
+    }
+
+    if ((e = getenv (DYAD_PATH_PRODUCER_ENV))) {
+        prod_managed_path = e;
+    } else {
+        prod_managed_path = NULL;
+    }
+
+    if ((e = getenv (DYAD_DTL_MODE_ENV))) {
+        dtl_mode_env_len = strlen (e);
+        if (strncmp (e, "FLUX_RPC", dtl_mode_env_len) == 0) {
+            dtl_mode = DYAD_DTL_FLUX_RPC;
+        } else if (strncmp (e, "UCX", dtl_mode_env_len) == 0) {
+            dtl_mode = DYAD_DTL_UCX;
+        } else {
+            if (debug) {
+                fprintf (stderr, "Invalid DTL mode provided through %s. \
+                        Defaulting to UCX\n", DYAD_DTL_MODE_ENV);
+            }
+            dtl_mode = DYAD_DTL_UCX;
```

Comment:
I think having UCX as default makes sense for the time being. Most systems nowadays have UCX, mainly because its used by MPI. Also, right now, the UCX dependency is required, so DYAD won't build without UCX.

We can think more about this later, but I think that should be a separate PR if we change it.

Reply:
libfabric is another to consider. AWS EFA is based on libfabric for example. So, we should definitely detect what is available and use it as the default instead of assuming UCX availability.

Reply:
This is can be a followup PR. Even if libfabric is not supported, UCX availability should be checked.

Reply:
Requiring UCX as the dependency is also problematic in the short term. Vanessa would not be able to try this if that is the case.

Reply:
Vanessa would just have to install UCX. UCX can work on cloud (although I can't say anything about performance). If a HPC-grade network (e.g., InfiniBand) is not present, UCX will just fall back to TCP/IP.